backend/app/controllers/system_messages.js

```diff
@@ -0,0 +1,48 @@
+const messageService = require('../services/messages');
+const logger = require('../logger');
+
+module.exports = {
+    send: (req, res) => {
+        if (
+            !req.body.to ||
+            !req.body.message
+        ) {
+            res.status(400).send('System message must have a \'to\' field and a \'message\' field').end();
+        } else {
+            messageService.sendSystemMessage(
+                req.app.get(messageService.SYSTEM_MESSAGE_USER_TOKEN_FIELD),
+                req.body.to,
+                req.body.message,
+                messageService.SYSTEM_MESSAGE_SUBJECT
+            )
+            .then(() => {
+                res.status(204).end();
+            })
+            .catch((err) => {
+                if (err.statusCode === 401) {
+                    logger.debug('Attempt to send a system message was unauthorized');
+                    logger.debug('Reauthenticating and trying again');
+                    return messageService.authAsSystemMessageUser()
```

Comment:
What happens if the reauthentication attempt fails?

Reply:
The `catch` on line 43 would get it and send back a 400 `'Failed to send system message'` to the client. If it was a temporary issue with the messaging service, the reauth should go through on a subsequent `/system_message` call. A totally crashed messaging service or an accidentally changed SYS_MESSAGE_USER password would just keep failing. I could add a Bologger call to log each failure so the issue could be found more easily? That would also depend on fixing Bologger.

Reply:
Got it. I think it would be ideal to have something more granular for error remediation.